src/astro/databases/sqlite.py

```diff
@@ -0,0 +1,116 @@
+
+DEFAULT_CONN_ID = "sqlite_default"
+
+
+class SqliteDatabase(BaseDatabase):
+    def __init__(self, conn_id: str = DEFAULT_CONN_ID):
+        if not conn_id:
+            conn_id = DEFAULT_CONN_ID
+        self.conn_id = conn_id
+
+    @property
+    def hook(self):
+        """
+        Retrieve Airflow hook to interface with the Sqlite database.
+        """
+        return SqliteHook(sqlite_conn_id=self.conn_id)
+
+    @property
+    def sqlalchemy_engine(self) -> Engine:
+        """
+        Return SQAlchemy engine.
+        """
+        uri = self.hook.get_uri()
+        if "////" not in uri:
+            uri = uri.replace("///", "////")
+        return create_engine(uri)
+
+    def run_sql(
+        self, sql_statement: Union[text, str], parameters: Optional[dict] = None
+    ) -> ResultProxy:
+        """
+        Run given SQL statement in the database using the Sqlalchemy engine.
+
+        :param sql_statement: SQL statement to be run on the engine
+        :param parameters: (optional) Parameters to be passed to the SQL statement
+        :return: Result of running the statement.
+        """
+        if parameters is None:
+            parameters = {}
+        return self.connection.execute(sql_statement, parameters)
+
+    # ---------------------------------------------------------
+    # Table metadata
+    # ---------------------------------------------------------
+    def get_table_qualified_name(self, table: Table) -> str:
+        """
+        Return the table qualified name.
+
+        :param table: The table we want to retrieve the qualified name for.
+        """
+        return str(table.name)
+
+    # ---------------------------------------------------------
+    # Load methods
+    # ---------------------------------------------------------
+    def load_file_to_table(
+        self,
+        source_file: str,  # TODO: replace by File object, which will contain normalization config
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ) -> None:
+        """
+        Upload the content of the source file to the target database.
+        If the table instance does not contain columns, this method automatically identify them using Pandas.
+
+        :param source_file: Path to original file (e.g. a "/tmp/sample_data.csv")
+        :param target_table: Details of the target table
+        :param if_exists: Strategy to be applied in case the target table exists
+        :param chunk_size: Specify the number of rows in each batch to be written at a time.
+        """
+        pandas_dataframe = load_file_into_dataframe(source_file)
+        self.load_pandas_dataframe_to_table(
+            pandas_dataframe, target_table, if_exists, chunk_size
+        )
+
+    def append_table(
+        self,
+        source_table: Table,
+        target_table: Table,
+        if_conflicts: AppendConflictStrategy = "exception",
+    ):
```

Comment:
Is there a reason we have append but not merge?

Reply:
@dimberman in a conversation we had while you were off, @kaxil, @utkarsharma2, and I discussed having a single function that would merge both features, with a different parameter `if_conflicts`. I thought this could be polemic - so I'd love your thoughts on this before further implementing it.